### PR TITLE
fix: 🐛 Redirect user back to service after modal closes

### PR DESCRIPTION
**Pull request by dBucik**
Fixed version redirect user back to the service after the modal window is closed. Previous implementation caused users to be stuck on opening the modal window informing them that the application is not processed and the service might redirect them back to this reg. form. However, this causes problems if users are redirected somewhere else, e.g. homepage of the service.

### CODE REVIEW

File: perun-wui-registrar/src/main/java/cz/metacentrum/perun/wui/registrar/pages/steps/SummaryStep.java
Comment (zlamalp) [inline perun-wui-registrar/src/main/java/cz/metacentrum/perun/wui/registrar/pages/steps/SummaryStep.java:870]: I'm not sure what this is supposed to do. Do you wish to perform redirect once you close the modal window? Because I think now you can try it multiple times and it always checks the current state of the submitted application and either shows the modal again or performs the redirect by passing click event to the default handler: `clickHandler.onClick(event);`.
Your code would simply add the handler to the button, hence when clicked again, it would show the modal window and perform the redirect together.

**Review by zlamalp — CHANGES_REQUESTED** [review]

File: perun-wui-registrar/src/main/java/cz/metacentrum/perun/wui/registrar/pages/steps/SummaryStep.java
Comment (dBucik) [reply perun-wui-registrar/src/main/java/cz/metacentrum/perun/wui/registrar/pages/steps/SummaryStep.java:870]: What I want to achieve is that even when the application is not processed yet, the user can be redirected to the target location, as the location might not be the service itself, just some "portal" with information. 
Perhaps an alternative and better solution would be to add another button to the modal window like "I understand and want to proceed" next to the "I understand" button.

File: perun-wui-registrar/src/main/java/cz/metacentrum/perun/wui/registrar/pages/steps/SummaryStep.java
Comment (zlamalp) [inline perun-wui-registrar/src/main/java/cz/metacentrum/perun/wui/registrar/pages/steps/SummaryStep.java:870]: Since we have 7s timeout before the redirect happens, I suggest to hide the modal (in my case message and spinner was hidden below the modal).

File: perun-wui-registrar/src/main/java/cz/metacentrum/perun/wui/registrar/pages/steps/SummaryStep.java
Comment (zlamalp) [inline perun-wui-registrar/src/main/java/cz/metacentrum/perun/wui/registrar/pages/steps/SummaryStep.java:866]: I probably wouldn't use PRIMARY button type, since this is sort of override for the default behavior, which is to wait for the approval. Now it looks like two very similar buttons.

Or if you think it will be better this way, switch what action is the default (both button type and their order in the modal window).

**Review by zlamalp — CHANGES_REQUESTED** [review]

File: perun-wui-registrar/src/main/java/cz/metacentrum/perun/wui/registrar/pages/steps/SummaryStep.java
Comment (dBucik) [reply perun-wui-registrar/src/main/java/cz/metacentrum/perun/wui/registrar/pages/steps/SummaryStep.java]: Modal should hide now.

File: perun-wui-registrar/src/main/java/cz/metacentrum/perun/wui/registrar/pages/steps/SummaryStep.java
Comment (dBucik) [reply perun-wui-registrar/src/main/java/cz/metacentrum/perun/wui/registrar/pages/steps/SummaryStep.java]: Swithched order and used `DEFAULT` for the "close but do not continue" button.

**Review by zlamalp — APPROVED** [review]

**Pull request merged**
